src/__tests__/Bugsnag.test.js

```diff
@@ -0,0 +1,190 @@
+beforeEach(() => {
+  // ensure freshly mocked clients in each test
+  jest.resetModules()
+
+  // mock globals set by RN at runtime
+  global.__DEV__ = false
+  global.ErrorUtils = {
+    setGlobalHandler: jest.fn(),
+    getGlobalHandler: jest.fn()
+  }
+})
+
+test('constructor(): throws if called with bad input', () => {
+  jest.mock('react-native', () => ({
+    NativeModules: { BugsnagReactNative: null }
+  }), { virtual: true })
+
+  const { Client } = require('../Bugsnag')
+
+  // non-string and non-undefined value as config/api key
+  expect(() => new Client(1)).toThrowError(/API key/)
+
+  // missing BugsnagReactNative bindings
+  expect(() => new Client('API_KEY')).toThrowError(/BugsnagReactNative/)
+})
+
+test('constructor(): works if everything is ok', () => {
+  const mockStart = jest.fn()
+  jest.mock('react-native', () => ({
+    NativeModules: { BugsnagReactNative: { startWithOptions: mockStart } }
+  }), { virtual: true })
+
+  const { Client } = require('../Bugsnag')
+  const c = new Client('API_KEY')
+  expect(mockStart).toHaveBeenCalledWith(expect.objectContaining({ apiKey: 'API_KEY' }))
+  expect(global.ErrorUtils.getGlobalHandler).toHaveBeenCalled()
+  expect(global.ErrorUtils.setGlobalHandler).toHaveBeenCalled()
+})
+
+test('handleUncaughtErrors(): error handler calls notify(…) correctly', () => {
+  jest.mock('react-native', () => ({
+    NativeModules: { BugsnagReactNative: { startWithOptions: jest.fn() } }
+  }), { virtual: true })
+
+  const { Client } = require('../Bugsnag')
+  const c = new Client('API_KEY')
+
+  // get the function that the client passed to the global error handler
+  const handler = global.ErrorUtils.setGlobalHandler.mock.calls[0][0]
+
+  c.notify = jest.fn()
+  handler(new Error('boom!'), false)
+  expect(c.notify).toHaveBeenCalledWith(expect.any(Error), expect.any(Function), false, expect.any(Function))
+})
+
+test('handlePromiseRejections(): error handler calls notify(…) correctly', () => {
+  jest.mock('react-native', () => ({
+    NativeModules: { BugsnagReactNative: { startWithOptions: jest.fn() } }
+  }), { virtual: true })
+
+  const { Client, Configuration } = require('../Bugsnag')
+  const config = new Configuration('API_KEY')
+  config.handlePromiseRejections = true
+  const c = new Client(config)
+
+  c.notify = jest.fn()
+  Promise.reject(new Error('pboom!'))
+  // promise rejection handler has a slight async delay before running, so check after minimal timeout
+  setTimeout(() => expect(c.notify).toHaveBeenCalledWith(expect.any(Error)))
```

Comment:
I think this actually just causes the assertion to happen after the test has finished running. Jest tests appear to be synchronous unless you return a promise 🤔